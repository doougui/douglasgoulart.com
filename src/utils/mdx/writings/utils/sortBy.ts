import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
