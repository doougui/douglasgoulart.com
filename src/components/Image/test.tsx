import { fireEvent, render, screen, waitFor } from 'utils/tests';

import { Image } from '.';

describe('<Image />', () => {
  it('should render correctly', async () => {
    const src = 'https://picsum.photos/200';
    const generatedSrc =
      '/_next/image?url=https%3A%2F%2Fpicsum.photos%2F200&w=640&q=75';

    const { container } = render(
      <Image src={src} alt="Image" width={200} height={200} />,
    );

    const img = screen.getByAltText('Image');

    expect(img).toBeInTheDocument();

    await waitFor(() => {
      expect(img).toHaveAttribute('src', generatedSrc);
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should use fallback on error', async () => {
    const src = 'https://picsum.photos/200';
    const generatedSrc =
      '/_next/image?url=https%3A%2F%2Fpicsum.photos%2F400&w=640&q=75';

    render(
      <Image
        src={src}
        fallback="https://picsum.photos/400"
        alt="Image"
        width={200}
        height={100}
      />,
    );

    const img = screen.getByAltText('Image');

    fireEvent.error(img);

    await waitFor(() => {
      expect(img).toHaveAttribute('src', generatedSrc);
    });
  });
});
