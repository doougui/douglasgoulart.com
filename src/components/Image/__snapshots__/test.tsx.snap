// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render correctly 1`] = `
<span
  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
>
  <span
    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
  >
    <img
      alt=""
      aria-hidden="true"
      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27200%27%20height=%27200%27/%3e"
      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
    />
  </span>
  <img
    alt="Image"
    data-nimg="intrinsic"
    decoding="async"
    src="/_next/image?url=https%3A%2F%2Fpicsum.photos%2F200&w=640&q=75"
    srcset="/_next/image?url=https%3A%2F%2Fpicsum.photos%2F200&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fpicsum.photos%2F200&w=640&q=75 2x"
    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
  />
  <noscript />
</span>
`;
